src/styles/index.scss

```diff
@@ -1,3 +1,62 @@
-body {
+@import './utils/variables';
+
+* {
+  box-sizing: border-box;
   margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
+body {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: $padding;
+  width: calc(
+    #{$size-of-cell} * #{$number-of-columns} + #{$gap} * 6 + #{$padding} * 2
+  );
+
+  &__day {
+    height: 100px;
+    width: 100px;
+    border: 1px solid #000;
+    background-color: $color-of-cell;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    transition: all $duration-of-animation;
+
+    &:hover {
+      cursor: pointer;
+      background-color: $color-on-hover;
+      transform: translateY(-20px);
+    }
+  }
+
+  @for $days from 1 through 31 {
+    &__day:nth-child(#{$days})::before {
+      content: '#{$days}';
+    }
+  }
```

Comment:
The loop to generate the days inside the calendar can be simplified by using a mixin for creating each day and then applying it inside the loop. This will make the code more maintainable and easier to read.